### ISSUE TITLE
Out of memory when converting rebar (example file)

### ISSUE THREAD

**Issue by johanrd** [IfcOpenShell/IfcOpenShell#1005]
1. Convert [1005--reinforcingBar--outOfMemory--augmented.ifc](https://github.com/IfcOpenShell/files/blob/master/1005--reinforcingBar--outOfMemory--augmented.ifc) with with 55a5409 (OCCT 7.3.0p4 / 7.4.0p1 / 7.3.0), and experience out of memory error code:

```
FATAL: An error occurred, status code 137.
```

When converting with 747ddeb20cc9083ba4281d7ddb283ac0e1bd0e12 (v0.5.0) or 42e0058 (0.6.0 Feb. 2019), the output is no geometry:
```
[Notice] include entities IfcCovering IfcFlowSegment IfcReinforcingBar
[Notice] Using world coords when writing WaveFront OBJ files
[Error] {2AerVx3fDfxrXJreBzNtNR} Adjusted edge end-point with distance 2.736287091669832e-07 on:
#30664=IfcCompositeCurve((#30588,#30595,#30599,#30603,#30609,#30614,#30621,#30627,#30634,#30640,#30646,#30651,#30657,#30663),.F.)
[Error] {2AerVx3fDfxrXJreBzNtNR} Error in: BRepAdaptor_Curve::No geometry
Failed to convert:
#30665=IfcSweptDiskSolid(#30664,7.,$,0.,1269.66106237146)
[Error] {2AerVx3fDfxrXJreBzNtNR} Adjusted edge end-point with distance 2.7362870916697579e-07 on:
#30705=IfcCompositeCurve((#30673,#30675,#30677,#30680,#30682,#30684,#30686,#30688,#30634,#30692,#30694,#30698,#30700,#30704),.F.)
[Error] {2AerVx3fDfxrXJreBzNtNR} Error in: BRepAdaptor_Curve::No geometry
Failed to convert:
#30706=IfcSweptDiskSolid(#30705,7.,$,0.,1269.66106237142)
[Error] {2AerVx3fDfxrXJreBzNtNR} Adjusted edge end-point with distance 2.7362870916697579e-07 on:
#30747=IfcCompositeCurve((#30716,#30718,#30720,#30722,#30724,#30726,#30728,#30730,#30732,#30736,#30738,#30742,#30744,#30746),.F.)
[Error] {2AerVx3fDfxrXJreBzNtNR} Error in: BRepAdaptor_Curve::No geometry
Failed to convert:
#30748=IfcSweptDiskSolid(#30747,7.,$,0.,1269.66106237141)
[Error] {2AerVx3fDfxrXJreBzNtNR} Adjusted edge end-point with distance 2.736287091669751e-07 on:
#30777=IfcCompositeCurve((#30716,#30718,#30756,#30758,#30760,#30762,#30764,#30766,#30732,#30768,#30738,#30772,#30774,#30776),.F.)
[Error] {2AerVx3fDfxrXJreBzNtNR} Error in: BRepAdaptor_Curve::No geometry
Failed to convert:
#30778=IfcSweptDiskSolid(#30777,7.,$,0.,1269.66106237144)
[Error] {2AerVx3fDfxrXJreBzNtNR} Adjusted edge end-point with distance 2.7362870916698188e-07 on:
#30809=IfcCompositeCurve((#30786,#30788,#30756,#30790,#30792,#30794,#30796,#30798,#30800,#30802,#30804,#30808,#30744,#30746),.F.)
[Error] {2AerVx3fDfxrXJreBzNtNR} Error in: BRepAdaptor_Curve::No geometry
Failed to convert:
#30810=IfcSweptDiskSolid(#30809,7.,$,0.,1269.66106237145)
[Error] No geometrical entities found
```

2. Expected geometry from Solibri:
<img width="415" alt="Screenshot 2020-09-18 at 12 36 29" src="https://user-images.githubusercontent.com/4601554/93588383-9bfd1600-f9ab-11ea-8821-6207573ac055.png">

**Comment by aothms** [comment]
Thanks yes, there are quite a bit of checks and balances during the composite- and trimmed curve generation. But a trimmed curve over a unbounded line trimmed by parameter values was not properly checked for degeneracy yet.

**Comment by johanrd** [comment]
Closing, as this seems to be fixed after bf57258
